robot/Cumulus/tests/browser/gift_entry/edit_template.robot

```diff
@@ -0,0 +1,140 @@
+*** Settings ***
+
+Resource        robot/Cumulus/resources/NPSP.robot
+Library         cumulusci.robotframework.PageObjects
+...             robot/Cumulus/resources/NPSPSettingsPageObject.py
+...             robot/Cumulus/resources/GiftEntryPageObject.py
+...             robot/Cumulus/resources/OpportunityPageObject.py
+...             robot/Cumulus/resources/PaymentPageObject.py
+Suite Setup     Run keywords
+...             Open Test Browser
+...             Enable Gift Entry
+...             Setup Test Data
+Suite Teardown  Capture Screenshot and Delete Records and Close Browser
+
+*** Variables ***
+&{method}         Default Value=Check
+&{check}          Default Value=abc11233
+&{date_default}   Default Value=Today
+&{custom}         Required=checked
+${amount}         50
+${msg}            Automation test
+
+*** Keywords ***
+Setup Test Data
+    &{contact} =     API Create Contact                 FirstName=${faker.first_name()}    LastName=${faker.last_name()}
+    Set suite variable    &{contact}
+    &{account} =     API Create Organization Account    Name=${faker.company()}
+    Set suite variable    &{account}
+    ${org_ns} =  Get Org Namespace Prefix
+    Set suite variable    ${org_ns}
+    ${ns} =  Get NPSP Namespace Prefix
+    Set suite variable    ${ns}
+    ${ui_date} =                     Get Current Date              result_format=%b %-d, %Y
+    Set suite variable    ${ui_date}
+    ${date} =     Get Current Date    result_format=%Y-%m-%d
+    Set suite variable    ${date}
+
+*** Test Cases ***
+
+Edit GE Template And Verify Changes
+    [Documentation]                  This test case makes edits to Default gift entry template and verifies that edits are saved 
+    ...                              and available when a Single gift or batch gift are created. 
+ 
+    [tags]                           feature:GE          W-039559   
+    # Edit Default template to add some default values and add a new field to form                   
+    Go To Page                       Landing                       GE_Gift_Entry
+    Click Link                       Templates
+    Select Template Action           Default Gift Entry Template   Edit
+    Current Page Should Be           Template                      GE_Gift_Entry
+    Click Gift Entry Button          Next: Form Fields
+    Select Object Group Field        Account 1                     custom_acc_text 
+    Select Object Group Field        Opportunity                   Donation Imported
+    Fill Template Form                      
+    ...                              Account 1: custom_acc_text=&{custom}
+    ...                              Opportunity: Close Date=&{date_default}
+    ...                              Payment: Check/Reference Number=&{check}
+    ...                              Payment: Payment Method=&{method} 
+    Click Gift Entry Button          Save & Close
```

Comment:
Can Fill_Template_form  keyword be repurposed  to  something like the below. That way you can specify multiple action criteria too. Writing just the pseudocode below
complete_action_on_template_form(**kwargs, action)
       Fill_template_form(*kwargs)
       if action.lower == "save & close":
               Click_gift_entry_button(action)
       if action.lower == "Save":
               click_button(action)

Reply:
I like to not include saving as part of the same keyword as there could be cases I just need to add something on form and see if causes some chain of actions and cancel out. If I include save or cancel as part of it, then I cant do such verification and also clicking on the save button is actually a next step even if we think of it in terms of manual steps.